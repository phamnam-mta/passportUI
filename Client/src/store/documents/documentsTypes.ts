export enum DocumentStatus {
    Loading = "Loading",
    Loaded = "Loaded",
    Analyzing = "Analyzing",
    Analyzed = "Analyzed",
    AnalyzeFailed = "AnalyzeFailed",
    Labeled = "Labeled",
}

export interface IDocumentStates {
    loadingStatus: DocumentStatus.Loaded | DocumentStatus.Loading;
    analyzingStatus?: DocumentStatus.Analyzed | DocumentStatus.Analyzing | DocumentStatus.AnalyzeFailed;
    labelingStatus?: DocumentStatus.Labeled;
}

export interface IRawDocument {
    name: string;
    type: string;
    url: string;
}

export interface IDocument extends IRawDocument {
    thumbnail: string;
    numPages: number;
    currentPage: number;
    states: IDocumentStates;
    expirationTime?: number;
}
