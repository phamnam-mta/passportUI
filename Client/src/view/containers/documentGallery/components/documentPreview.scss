@import "~@fluentui/react/dist/sass/References";

.documentcard-overlay {
    display: flex;
    justify-content: center;
    position: absolute;
    top: 0;
    height: 100px;
    width: 100%;
    background: rgba(243, 242, 241, 0.8);
}

.documentcard-status-badge {
    position: absolute;
    top: 0;
    right: 0;
    margin-top: 8px;
    margin-right: 4px;
    font-size: 16px;
    color: $ms-color-themePrimary;
}

.documentcard-badge {
    position: absolute;
    top: 0;
    right: 0;
    margin-top: 7px;
    margin-right: 4px;
}

.documentcard-delete-icon {
    position: absolute;
    bottom: 24px;
    right: 0;
    margin-right: 3px;
    color: $ms-color-gray130;
    font-size: 20px;
    padding: 4px;
    border-radius: 4px 0 0 0;
    background-color: rgba(255, 255, 255, 0.7);

    &:hover {
        color: $ms-color-themePrimary;
    }
}
