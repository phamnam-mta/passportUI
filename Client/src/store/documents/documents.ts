import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import { resetPredictions } from "store/predictions/predictions";
import { DocumentLoaderFactory, IDocumentLoader } from "utils/documentLoader";
import { DocumentStatus, IDocument, IRawDocument } from "./documentsTypes";

export type DocumentsState = { documents: IDocument[]; currentDocument: IDocument | null };

export const initialState: DocumentsState = { documents: [], currentDocument: null };

export const addDocuments = createAsyncThunk("documents/addDocuments", async (docsToAdd: IRawDocument[]) => {
    return Promise.all(docsToAdd.map(async (doc) => (await getLoader(doc)).loadDocumentMeta()));
});

export const setCurrentDocument = createAsyncThunk("documents/setCurrentDocument", async (document: IDocument) => {
    const loader = await getLoader(document);
    const documentPage = await loader.loadDocumentPage(document.currentPage);
    return { document, documentPage };
});

export const setCurrentPage = createAsyncThunk("documents/setCurrentPage", async (pageNumber: number, thunkApi) => {
    const { documents: state } = thunkApi.getState() as any;
    const loader = await getLoader(state.currentDocument);
    const documentPage = await loader.loadDocumentPage(pageNumber);
    return { pageNumber, documentPage };
});

const documentLoaders = new Map<string, IDocumentLoader>();

const getLoader = async (document: IRawDocument): Promise<IDocumentLoader> => {
    let loader = documentLoaders.get(document.url);
    if (!loader) {
        loader = await DocumentLoaderFactory.makeLoader(document);
        documentLoaders.set(document.url, loader);
    }

    return loader;
};

const documentsSlice = createSlice({
    name: "documents",
    initialState,
    reducers: {
        deleteDocument(state, action) {
            const docNameToDelete = action.payload.name;
            state.documents = state.documents.filter((document) => document.name !== docNameToDelete);
            documentLoaders.delete(action.payload.url);
        },
        setDocumentAnalyzingStatus(state, action) {
            const { name, status } = action.payload;
            const document = state.documents.find((document) => document.name === name);
            if (document) {
                document.states.analyzingStatus = status;
            }
            if (state.currentDocument?.name === name) {
                state.currentDocument!.states.analyzingStatus = status;
            }
        },
        setDocumentLabelingStatus(state, action) {
            const { name, status } = action.payload;
            const document = state.documents.find((document) => document.name === name);
            if (document) {
                document.states.labelingStatus = status;
            }
            if (state.currentDocument?.name === name) {
                state.currentDocument!.states.labelingStatus = status;
            }
        },
        clearCurrentDocument(state) {
            state.currentDocument = null;
        },
    },
    extraReducers: (builder) => {
        builder
            .addCase(resetPredictions, (state) => {
                state.documents.forEach((document) => {
                    document.states = { ...document.states, analyzingStatus: undefined };
                });
                if (state.currentDocument) {
                    state.currentDocument.states = { ...state.currentDocument.states, analyzingStatus: undefined };
                }
            })
            .addCase(addDocuments.pending, (state, action) => {
                const docs = action.meta.arg;
                docs.forEach((doc) => {
                    state.documents.push({
                        ...doc,
                        thumbnail: "",
                        numPages: 0,
                        currentPage: 0,
                        states: { loadingStatus: DocumentStatus.Loading },
                    });
                });
            })
            .addCase(setCurrentDocument.pending, (state, action) => {
                const document = action.meta.arg;
                const selectedDoc = state.documents.find((doc) => doc.name === document.name);
                if (selectedDoc) {
                    selectedDoc.states.loadingStatus = DocumentStatus.Loading;
                }
            })
            .addCase(setCurrentDocument.fulfilled, (state, action) => {
                const document: IDocument = {
                    ...action.payload.document,
                    states: {
                        ...action.payload.document.states,
                        loadingStatus: DocumentStatus.Loaded,
                    },
                };
                const selectedDoc = state.documents.find((doc) => doc.name === document.name);
                if (selectedDoc) {
                    selectedDoc.states.loadingStatus = DocumentStatus.Loaded;
                }
                state.currentDocument = document;
            })
            .addCase(setCurrentPage.fulfilled, (state, action) => {
                const { pageNumber } = action.payload;
                const document = state.documents.find((doc) => doc.name === state.currentDocument?.name);
                if (document) {
                    document.currentPage = pageNumber;
                }
                if (state.currentDocument) {
                    state.currentDocument.currentPage = pageNumber;
                }
            })
            .addCase(addDocuments.fulfilled, (state, action) => {
                const addedDocs = action.payload;
                const { documents } = state;
                addedDocs.forEach((doc) => {
                    const iDoc = documents.findIndex((d) => d.name === doc.name);
                    if (iDoc >= 0) {
                        documents[iDoc] = {
                            ...doc,
                            states: { ...documents[iDoc].states, loadingStatus: DocumentStatus.Loaded },
                        };
                    }
                });
            });
    },
});

export const { deleteDocument, setDocumentAnalyzingStatus, setDocumentLabelingStatus, clearCurrentDocument } =
    documentsSlice.actions;

export const reducer = documentsSlice.reducer;
