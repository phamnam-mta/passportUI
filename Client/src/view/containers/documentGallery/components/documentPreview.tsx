import * as React from "react";
import {
    IDocumentCardStyles,
    DocumentCard,
    DocumentCardImage,
    DocumentCardDetails,
    ImageFit,
    IDocumentCardTitleStyles,
    Spinner,
    SpinnerSize,
    FontIcon,
    Text,
} from "@fluentui/react";
import { Depths, NeutralColors } from "@fluentui/theme";

import { DocumentStatus, IDocumentStates } from "store/documents/documentsTypes";

import "./documentPreview.scss";

interface IDocumentPreviewProps {
    selected: boolean;
    documentName: string;
    documentStates: IDocumentStates;
    documentImageSrc: string;
    onDocumentClick: (documentTitle: string) => void;
    onDocumentDelete: (name: string) => void;
}

interface IDocumentPreviewState {
    isDocumentCardHovered: boolean;
}

type Props = IDocumentPreviewProps;

export class DocumentPreview extends React.PureComponent<Props, IDocumentPreviewState> {
    constructor(props: any) {
        super(props);
        this.state = {
            isDocumentCardHovered: false,
        };
    }

    private getCardStyles = (): IDocumentCardStyles => {
        return {
            root: {
                position: "relative",
                height: 124,
                minWidth: 124,
                width: 124,
                border: this.props.selected ? "1px solid " + NeutralColors.black : "none",
                borderRadius: "2px",
                boxShadow: Depths.depth4,
                boxSizing: "border-box",
                marginLeft: 16,
                marginRight: "auto",
            },
        };
    };

    private titleStyles: IDocumentCardTitleStyles = {
        /*
            Set width to make shouldTruncate work correctly
            Leave some width buffer to overflow, since the calculation of shouldTruncate might not be completely fit.
            Set overflow to visible and make nowrap to display the overflow part.
        */
        root: {
            fontSize: "12px",
            overflow: "hidden",
            whiteSpace: "nowrap",
            textOverflow: "ellipsis",
            height: 20,
            lineHeight: 20,
            width: 110, // Max 124.
            padding: 0,
            textAlign: "center",
        },
    };

    private getDocumentBadge = () => {
        const { documentStates } = this.props;
        if (documentStates.labelingStatus === DocumentStatus.Labeled) {
            return <FontIcon title="Labeled" iconName="CircleFill" style={{ color: "#6C179A", fontSize: 14 }} />;
        } else if (documentStates.analyzingStatus === DocumentStatus.Analyzed) {
            return <FontIcon title="Analyzed" iconName="CircleFill" style={{ color: "#0F703B", fontSize: 14 }} />;
        }
    };

    private handleClick = () => {
        const { onDocumentClick, documentName } = this.props;
        onDocumentClick(documentName);
    };

    private handleDocumentDelete = (evt) => {
        evt.stopPropagation();
        const { onDocumentDelete, documentName } = this.props;
        onDocumentDelete(documentName);
    };

    private getAnalyzeStatusString = () => {
        const { documentStates } = this.props;
        switch (documentStates.analyzingStatus) {
            case DocumentStatus.Analyzed:
                return "Analyzed";
            case DocumentStatus.Analyzing:
                return "Analyzing...";
            case DocumentStatus.AnalyzeFailed:
                return "Analyze failed";
            default:
                return "Loaded";
        }
    };

    public render() {
        const { documentName, documentStates, documentImageSrc } = this.props;
        const { isDocumentCardHovered } = this.state;

        return (
            <DocumentCard
                key={documentName}
                title={documentName}
                tabIndex={0}
                aria-label={documentName + "Document Status:" + this.getAnalyzeStatusString()}
                styles={this.getCardStyles()}
                onClick={this.handleClick}
                onMouseEnter={() => {
                    this.setState({ isDocumentCardHovered: true });
                }}
                onMouseLeave={() => {
                    this.setState({ isDocumentCardHovered: false });
                }}
            >
                <DocumentCardImage height={100} imageFit={ImageFit.contain} imageSrc={documentImageSrc} />
                <DocumentCardDetails styles={{ root: { alignItems: "center" } }}>
                    <Text title={documentName} styles={this.titleStyles}>
                        {documentName}
                    </Text>
                    {isDocumentCardHovered && (
                        <FontIcon
                            className="documentcard-delete-icon"
                            title="Delete"
                            iconName="delete"
                            onClick={this.handleDocumentDelete}
                        />
                    )}
                </DocumentCardDetails>
                {documentStates.analyzingStatus === DocumentStatus.Analyzing && (
                    <div className="documentcard-overlay">
                        <Spinner
                            className="analyze-spinner"
                            size={SpinnerSize.medium}
                            label="Analyzing..."
                            styles={{ label: { fontSize: 14 } }}
                        />
                    </div>
                )}
                <div className="documentcard-badge">{this.getDocumentBadge()}</div>
            </DocumentCard>
        );
    }
}

export default DocumentPreview;
