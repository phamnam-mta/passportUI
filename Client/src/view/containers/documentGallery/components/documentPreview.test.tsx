import { shallow } from "enzyme";
import * as React from "react";
import { DocumentStatus } from "store/documents/documentsTypes";
import { DocumentPreview } from "./documentPreview";

describe("<DocumentPreview />", () => {
    let baseProps;

    beforeEach(() => {
        baseProps = {
            selected: false,
            documentName: "title",
            documentStates: { loadingStatus: DocumentStatus.Loaded },
            documentImageSrc: "sample.jpg",
            onDocumentClick: jest.fn(),
            onDocumentDelete: jest.fn(),
        };
    });

    describe("Rendering", () => {
        it("should match snapshot.", () => {
            const wrapper = shallow(<DocumentPreview {...baseProps} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when the document is selected", () => {
            const props = {
                ...baseProps,
                selected: true,
            };

            const wrapper = shallow(<DocumentPreview {...props} />);
            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when the document is analyzing", () => {
            const props = {
                ...baseProps,
                documentStates: { ...baseProps.documentStates, analyzingStatus: DocumentStatus.Analyzing },
            };

            const wrapper = shallow(<DocumentPreview {...props} />);
            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when the document is analyzed", () => {
            const props = {
                ...baseProps,
                documentStates: { ...baseProps.documentStates, analyzingStatus: DocumentStatus.Analyzed },
            };

            const wrapper = shallow(<DocumentPreview {...props} />);
            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when the document is labeled", () => {
            const props = {
                ...baseProps,
                documentStates: {
                    ...baseProps.documentStates,
                    analyzingStatus: DocumentStatus.Analyzed,
                    labelingStatus: DocumentStatus.Labeled,
                },
            };

            const wrapper = shallow(<DocumentPreview {...props} />);
            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot with delete Icon, when mouse enter document card", () => {
            const wrapper = shallow(<DocumentPreview {...baseProps} />) as any;
            const documentCard = wrapper.find("StyledDocumentCardBase").first();
            documentCard.simulate("mouseenter");

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Event Handling", () => {
        it("should call onDocumentClick when clicked on element", () => {
            const wrapper = shallow(<DocumentPreview {...baseProps} />);
            wrapper.simulate("click");

            expect(baseProps.onDocumentClick).toBeCalledWith(baseProps.documentName);
        });

        it("should set isDocumentCardHovered to be true, when mouse enter document card", () => {
            const wrapper = shallow(<DocumentPreview {...baseProps} />) as any;
            const documentCard = wrapper.find("StyledDocumentCardBase").first();
            documentCard.simulate("mouseenter");

            expect(wrapper.state("isDocumentCardHovered")).toBe(true);
        });

        it("should set isDocumentCardHovered to be false, when mouse leave document card", () => {
            const wrapper = shallow(<DocumentPreview {...baseProps} />) as any;
            wrapper.setState({ isDocumentCardHovered: true });
            const documentCard = wrapper.find("StyledDocumentCardBase").first();
            documentCard.simulate("mouseleave");

            expect(wrapper.state("isDocumentCardHovered")).toBe(false);
        });

        it("should delete document and not setCurrentDocument to it, when delete icon is clicked", () => {
            const clickEvent = new Event("click");
            const wrapper = shallow(<DocumentPreview {...baseProps} />) as any;
            wrapper.setState({ isDocumentCardHovered: true });
            const deleteIcon = wrapper.find(".documentcard-delete-icon").first();
            deleteIcon.simulate("click", clickEvent);

            expect(baseProps.onDocumentDelete).toBeCalledTimes(1);
            expect(baseProps.onDocumentDelete).toBeCalledWith(baseProps.documentName);
            expect(baseProps.onDocumentClick).not.toBeCalled();
        });
    });
});
