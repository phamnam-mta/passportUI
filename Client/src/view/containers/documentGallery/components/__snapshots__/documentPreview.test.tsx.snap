// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DocumentPreview /> Rendering should match snapshot with delete Icon, when mouse enter document card 1`] = `
<StyledDocumentCardBase
  aria-label="titleDocument Status:Loaded"
  key="title"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  styles={
    Object {
      "root": Object {
        "border": "none",
        "borderRadius": "2px",
        "boxShadow": "0 1.6px 3.6px 0 rgba(0, 0, 0, 0.132), 0 0.3px 0.9px 0 rgba(0, 0, 0, 0.108)",
        "boxSizing": "border-box",
        "height": 124,
        "marginLeft": 16,
        "marginRight": "auto",
        "minWidth": 124,
        "position": "relative",
        "width": 124,
      },
    }
  }
  tabIndex={0}
  title="title"
>
  <StyledDocumentCardImageBase
    height={100}
    imageFit={1}
    imageSrc="sample.jpg"
  />
  <StyledDocumentCardDetailsBase
    styles={
      Object {
        "root": Object {
          "alignItems": "center",
        },
      }
    }
  >
    <Text
      styles={
        Object {
          "root": Object {
            "fontSize": "12px",
            "height": 20,
            "lineHeight": 20,
            "overflow": "hidden",
            "padding": 0,
            "textAlign": "center",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
            "width": 110,
          },
        }
      }
      title="title"
    >
      title
    </Text>
    <FontIcon
      className="documentcard-delete-icon"
      iconName="delete"
      onClick={[Function]}
      title="Delete"
    />
  </StyledDocumentCardDetailsBase>
  <div
    className="documentcard-badge"
  />
</StyledDocumentCardBase>
`;

exports[`<DocumentPreview /> Rendering should match snapshot, when the document is analyzed 1`] = `
<StyledDocumentCardBase
  aria-label="titleDocument Status:Analyzed"
  key="title"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  styles={
    Object {
      "root": Object {
        "border": "none",
        "borderRadius": "2px",
        "boxShadow": "0 1.6px 3.6px 0 rgba(0, 0, 0, 0.132), 0 0.3px 0.9px 0 rgba(0, 0, 0, 0.108)",
        "boxSizing": "border-box",
        "height": 124,
        "marginLeft": 16,
        "marginRight": "auto",
        "minWidth": 124,
        "position": "relative",
        "width": 124,
      },
    }
  }
  tabIndex={0}
  title="title"
>
  <StyledDocumentCardImageBase
    height={100}
    imageFit={1}
    imageSrc="sample.jpg"
  />
  <StyledDocumentCardDetailsBase
    styles={
      Object {
        "root": Object {
          "alignItems": "center",
        },
      }
    }
  >
    <Text
      styles={
        Object {
          "root": Object {
            "fontSize": "12px",
            "height": 20,
            "lineHeight": 20,
            "overflow": "hidden",
            "padding": 0,
            "textAlign": "center",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
            "width": 110,
          },
        }
      }
      title="title"
    >
      title
    </Text>
  </StyledDocumentCardDetailsBase>
  <div
    className="documentcard-badge"
  >
    <FontIcon
      iconName="CircleFill"
      style={
        Object {
          "color": "#0F703B",
          "fontSize": 14,
        }
      }
      title="Analyzed"
    />
  </div>
</StyledDocumentCardBase>
`;

exports[`<DocumentPreview /> Rendering should match snapshot, when the document is analyzing 1`] = `
<StyledDocumentCardBase
  aria-label="titleDocument Status:Analyzing..."
  key="title"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  styles={
    Object {
      "root": Object {
        "border": "none",
        "borderRadius": "2px",
        "boxShadow": "0 1.6px 3.6px 0 rgba(0, 0, 0, 0.132), 0 0.3px 0.9px 0 rgba(0, 0, 0, 0.108)",
        "boxSizing": "border-box",
        "height": 124,
        "marginLeft": 16,
        "marginRight": "auto",
        "minWidth": 124,
        "position": "relative",
        "width": 124,
      },
    }
  }
  tabIndex={0}
  title="title"
>
  <StyledDocumentCardImageBase
    height={100}
    imageFit={1}
    imageSrc="sample.jpg"
  />
  <StyledDocumentCardDetailsBase
    styles={
      Object {
        "root": Object {
          "alignItems": "center",
        },
      }
    }
  >
    <Text
      styles={
        Object {
          "root": Object {
            "fontSize": "12px",
            "height": 20,
            "lineHeight": 20,
            "overflow": "hidden",
            "padding": 0,
            "textAlign": "center",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
            "width": 110,
          },
        }
      }
      title="title"
    >
      title
    </Text>
  </StyledDocumentCardDetailsBase>
  <div
    className="documentcard-overlay"
  >
    <StyledSpinnerBase
      className="analyze-spinner"
      label="Analyzing..."
      size={2}
      styles={
        Object {
          "label": Object {
            "fontSize": 14,
          },
        }
      }
    />
  </div>
  <div
    className="documentcard-badge"
  />
</StyledDocumentCardBase>
`;

exports[`<DocumentPreview /> Rendering should match snapshot, when the document is labeled 1`] = `
<StyledDocumentCardBase
  aria-label="titleDocument Status:Analyzed"
  key="title"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  styles={
    Object {
      "root": Object {
        "border": "none",
        "borderRadius": "2px",
        "boxShadow": "0 1.6px 3.6px 0 rgba(0, 0, 0, 0.132), 0 0.3px 0.9px 0 rgba(0, 0, 0, 0.108)",
        "boxSizing": "border-box",
        "height": 124,
        "marginLeft": 16,
        "marginRight": "auto",
        "minWidth": 124,
        "position": "relative",
        "width": 124,
      },
    }
  }
  tabIndex={0}
  title="title"
>
  <StyledDocumentCardImageBase
    height={100}
    imageFit={1}
    imageSrc="sample.jpg"
  />
  <StyledDocumentCardDetailsBase
    styles={
      Object {
        "root": Object {
          "alignItems": "center",
        },
      }
    }
  >
    <Text
      styles={
        Object {
          "root": Object {
            "fontSize": "12px",
            "height": 20,
            "lineHeight": 20,
            "overflow": "hidden",
            "padding": 0,
            "textAlign": "center",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
            "width": 110,
          },
        }
      }
      title="title"
    >
      title
    </Text>
  </StyledDocumentCardDetailsBase>
  <div
    className="documentcard-badge"
  >
    <FontIcon
      iconName="CircleFill"
      style={
        Object {
          "color": "#6C179A",
          "fontSize": 14,
        }
      }
      title="Labeled"
    />
  </div>
</StyledDocumentCardBase>
`;

exports[`<DocumentPreview /> Rendering should match snapshot, when the document is selected 1`] = `
<StyledDocumentCardBase
  aria-label="titleDocument Status:Loaded"
  key="title"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  styles={
    Object {
      "root": Object {
        "border": "1px solid #000000",
        "borderRadius": "2px",
        "boxShadow": "0 1.6px 3.6px 0 rgba(0, 0, 0, 0.132), 0 0.3px 0.9px 0 rgba(0, 0, 0, 0.108)",
        "boxSizing": "border-box",
        "height": 124,
        "marginLeft": 16,
        "marginRight": "auto",
        "minWidth": 124,
        "position": "relative",
        "width": 124,
      },
    }
  }
  tabIndex={0}
  title="title"
>
  <StyledDocumentCardImageBase
    height={100}
    imageFit={1}
    imageSrc="sample.jpg"
  />
  <StyledDocumentCardDetailsBase
    styles={
      Object {
        "root": Object {
          "alignItems": "center",
        },
      }
    }
  >
    <Text
      styles={
        Object {
          "root": Object {
            "fontSize": "12px",
            "height": 20,
            "lineHeight": 20,
            "overflow": "hidden",
            "padding": 0,
            "textAlign": "center",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
            "width": 110,
          },
        }
      }
      title="title"
    >
      title
    </Text>
  </StyledDocumentCardDetailsBase>
  <div
    className="documentcard-badge"
  />
</StyledDocumentCardBase>
`;

exports[`<DocumentPreview /> Rendering should match snapshot. 1`] = `
<StyledDocumentCardBase
  aria-label="titleDocument Status:Loaded"
  key="title"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  styles={
    Object {
      "root": Object {
        "border": "none",
        "borderRadius": "2px",
        "boxShadow": "0 1.6px 3.6px 0 rgba(0, 0, 0, 0.132), 0 0.3px 0.9px 0 rgba(0, 0, 0, 0.108)",
        "boxSizing": "border-box",
        "height": 124,
        "marginLeft": 16,
        "marginRight": "auto",
        "minWidth": 124,
        "position": "relative",
        "width": 124,
      },
    }
  }
  tabIndex={0}
  title="title"
>
  <StyledDocumentCardImageBase
    height={100}
    imageFit={1}
    imageSrc="sample.jpg"
  />
  <StyledDocumentCardDetailsBase
    styles={
      Object {
        "root": Object {
          "alignItems": "center",
        },
      }
    }
  >
    <Text
      styles={
        Object {
          "root": Object {
            "fontSize": "12px",
            "height": 20,
            "lineHeight": 20,
            "overflow": "hidden",
            "padding": 0,
            "textAlign": "center",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
            "width": 110,
          },
        }
      }
      title="title"
    >
      title
    </Text>
  </StyledDocumentCardDetailsBase>
  <div
    className="documentcard-badge"
  />
</StyledDocumentCardBase>
`;
